app/src/main/java/com/github/onedirection/navigation/fragment/sign/SignViewModel.java

```diff
@@ -0,0 +1,60 @@
+package com.github.onedirection.navigation.fragment.sign;
+
+import android.util.Patterns;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+import com.github.onedirection.R;
+import com.github.onedirection.authentication.FirebaseAuthentication;
+import com.github.onedirection.authentication.User;
+
+public class SignViewModel extends ViewModel {
+
+    private final MutableLiveData<SignFormState> signFormState = new MutableLiveData<>();
+    private final MutableLiveData<User> userResult = new MutableLiveData<>();
+
+    LiveData<SignFormState> getLoginFormState() {
+        return signFormState;
+    }
+
+    LiveData<User> getUserResult() {
+        return userResult;
+    }
+
+    public void sign(String username, String password, boolean register) {
+        FirebaseAuthentication auth = FirebaseAuthentication.getInstance();
```

Comment:
Would be better to use
```suggestion
        AuthenticationService auth = AuthenticationService.getDefaultInstance();
```
so there is no direct dependency on the FirebaseAuthentication.

Reply:
True, I've change it